firedantic/models.py

```diff
@@ -0,0 +1,89 @@
+from abc import ABC
+from typing import List, Optional, Type, TypeVar
+
+import pydantic
+from firedantic.configurations import CONFIGURATIONS
+from firedantic.exceptions import ModelNotFoundError
+from google.cloud.firestore_v1 import CollectionReference, DocumentReference
+
+TModel = TypeVar("TModel", bound="Model")
+
+
+class Model(pydantic.BaseModel, ABC):
+    """Base model class.
+
+    Implements basic functionality for Pydantic models, such as save, delete, find etc.
+    """
+
+    __collection__: Optional[str] = None
+
+    id: Optional[str] = None
+
+    def save(self) -> None:
+        """Saves this model in the database."""
+        data = self.dict()
+        if "id" in data:
+            del data["id"]
+
+        doc_ref = self._get_doc_ref()
+        doc_ref.set(data)
+        self.id = doc_ref.id
+
+    def delete(self) -> None:
+        """Deletes this model from the database."""
+        self._get_doc_ref().delete()
+
+    @classmethod
+    def find(cls: Type[TModel], filter_: dict) -> List[TModel]:
+        """Returns a list of models from the database based on a filter.
+
+        :param filter_: The filter criteria.
```

Comment:
Since this is a library, I think it would be nice to give an example of a query. Either here or in the README